### PR TITLE
OKTA-748810 - Crash/Non-fatal - <CF_IS_OBJC Firebase>

### PR DESCRIPTION
**Pull request by IldarAbdullin-okta**
Check results of `CFHTTPMessageCopyRequestMethod` and `CFHTTPMessageCopyRequestURL` for nil

### CODE REVIEW

File: Sources/AppAuth/macOS/OKTRedirectHTTPHandler.m [hunk 1]
```diff
@@ -127,11 +127,22 @@ - (void)stopHTTPListener {
   _httpServ = nil;
 }
 
+- (BOOL)isOptionsHTTPServerRequest:(HTTPServerRequest *)request {
+    CFStringRef method = CFHTTPMessageCopyRequestMethod(request.request);
+    if (method == nil) {
+        return NO;
+    }
+
+    BOOL isOptionsRequest = CFStringCompare(method, (__bridge CFStringRef)@"OPTIONS", 0) == kCFCompareEqualTo;
```
Comment (pavelkozlov-okta) [inline Sources/AppAuth/macOS/OKTRedirectHTTPHandler.m:136]: I'd suggest using `kCFCompareCaseInsensitive` option for CFStringCompare (just like `CFHTTPMessageCopyRequestMethod` does internally for `GET` method).

File: Sources/AppAuth/macOS/OKTRedirectHTTPHandler.m [hunk 2]
```diff
@@ -127,11 +127,22 @@ - (void)stopHTTPListener {
...
+}
+
 - (void)HTTPConnection:(HTTPConnection *)conn didReceiveRequest:(HTTPServerRequest *)mess {
   // Handle private network preflight
   // https://developer.chrome.com/blog/private-network-access-preflight/
-  CFStringRef method = CFHTTPMessageCopyRequestMethod(mess.request);
-  BOOL isOptionsRequest = CFStringCompare(method, (__bridge CFStringRef)@"OPTIONS", 0) == kCFCompareEqualTo;
+  BOOL isOptionsRequest = [self isOptionsHTTPServerRequest:mess];
 
   CFStringRef requestPrivateNetwork = isOptionsRequest ? CFHTTPMessageCopyHeaderFieldValue(mess.request, (__bridge CFStringRef)@"Access-Control-Request-Private-Network") : nil;
```
Comment (pavelkozlov-okta) [inline Sources/AppAuth/macOS/OKTRedirectHTTPHandler.m:147]: Seems like `requestPrivateNetwork` can also leak after calling `CFHTTPMessageCopyHeaderFieldValue`

File: Sources/AppAuth/macOS/OKTRedirectHTTPHandler.m (hunk 1 above)
Comment (IldarAbdullin-okta) [reply Sources/AppAuth/macOS/OKTRedirectHTTPHandler.m:136]: That makes sense. Wanted to introduce minimal changes to the existing logic though. But we can cover this in unit tests, lemme do this the change

Comment (IldarAbdullin-okta) [reply Sources/AppAuth/macOS/OKTRedirectHTTPHandler.m]: Addressed. Added more tests

**Review by pavelkozlov-okta — APPROVED** [review]

**Review by FeiChen-okta — APPROVED** [review]

**Review by moisesolmedo-okta — APPROVED** [review]

**Review by pavelkozlov-okta — APPROVED** [review]

**Review by pavelkozlov-okta — APPROVED** [review]

**Pull request merged**